.drone.yml

```diff
@@ -1,129 +1,128 @@
 pipeline:
   test_build:
-    image: docker:1.11
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - apk add --no-cache bash sed grep bc coreutils
+      - apk add --no-cache bash sed grep bc coreutils curl
       - ./ci-build.sh
     when:
-      event: [push, pull_request, tag]
+      event: [push, pull_request]
   
   build_clamav:
-    image: docker:1.11
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker build -t clamav .
+      - docker build -t clamav:$${DRONE_COMMIT_SHA} .
     when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
+      branch: master
+      event: push
   
   build_clamav-rest:
-    image: docker:1.11
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker build -t clamav-rest clamav-rest
+      - docker build -t clamav-rest:$${DRONE_COMMIT_SHA} clamav-rest
     when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
-  
-  clamav_to_quay:
-    image: docker:1.11
+      branch: master
+      event: push
+ 
+  build_clamav:
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker login -u="ukhomeofficedigital+drone" -p=${CLAMAV_REPO_TOKEN} quay.io
-      - docker tag clamav quay.io/ukhomeofficedigital/clamav:${DRONE_COMMIT_SHA}
-      - docker push quay.io/ukhomeofficedigital/clamav:${DRONE_COMMIT_SHA}
+      - docker build -t clamav:$${DRONE_COMMIT_SHA} .
     when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
+      event: tag
   
-  clamav-rest_to_quay:
-    image: docker:1.11
+  build_clamav-rest:
```

Comment:
This is duplicate of step above. You can use multiple events if you need that to trigger on push and tag.

Reply:
You can but then you wouldn't be able to specify the master branch (for the push event). So either you have 2 steps each (as it is now) or remove the branch condition so that it triggers the pipeline on every push on every branch (as well as triggering the test build pipeline). I felt that this way was better.